agent/web_reach.py

```
@@ -0,0 +1,428 @@
+                    current = nxt
+                    continue
+                resp.raise_for_status()
+                return resp
+        raise ValueError("too many redirects")
+
+    def _fetch_direct(self, url: str) -> dict[str, Any]:
+        """httpx-only fallback used when the .github/scripts helpers can't be
+        loaded (e.g. a deploy image that excludes that directory)."""
+        try:
+            resp = self._safe_get(url)
+        except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)